### DISCUSSION TITLE
City Dropdown in New Checkout Page

### DISCUSSION THREAD

**Discussion by ahmadiba** [woocommerce/woocommerce#53539] [Checkout Flow]
Hello, good day.

In the new version of WooCommerce and with the addition of the new checkout page to the store, how can the city field be changed to a dropdown field? In the old version of the checkout page, this problem was solved with code, but the code that worked for the old version no longer works. Please guide us on how to fix this issue. We want the list of cities to be based on the selected province from the list of provinces.

**Comment by d7428** [comment]
I have the same question.

With the Classic Checkout, it is very easy to change the shipping_city field into a dropdown containing a small, predefined list of cities, each with its own fixed shipping cost.

We are florists offering our own local delivery service within a small geographical area.

This necessity can also be applied to other local businesses, such as restaurants and other delivery services for various goods.

### Is this functionality even theoretically possible using the new Block Checkout?

**Comment by geekontheroad** [comment]
I'm also looking for this functionality for the city field. However, I need more than just converting it into a select field—I need to dynamically control which cities are displayed based on the selected state.

Additionally, I want to manage the options of custom fields added to the checkout page using `woocommerce_register_additional_checkout_field`. While this function allows us to configure options once, I’m looking for a way to update them dynamically on the frontend.

My current use case involves a Thai address format, which consists of three dependent select fields: Province (State), District (City), and Subdistrict. Each field should update dynamically based on the selection in the previous one.

Does anyone have insights on how to achieve this efficiently with the block editor? This site is currently still using the old shortcodes because of this limitation.
